### PR TITLE
added commands module

### PR DESCRIPTION
**Pull request by tingey21**
## PR Summary

This PR addresses the issue of custom Vim commands not being directly exposed or easily accessible. To tackle this, we've introduced a streamlined approach to enhance command handling:

- **Added a New File**: Incorporates command handling logic.
- **Implemented Helper Functions**: Simplifies command registration and management.
- **Enhanced Command Exposure**: `M.commands` now effectively exposes commands to Vim, allowing for straightforward integration.
- **Documented in the Readme**: Extend `M.commands` to introduce additional functionality.

### Key Benefits

- **Ease of Adding New Commands**: Extend `M.commands` to introduce additional functionality.
- **Improved Command Management**: Centralizes command logic for better maintainability.
- **Direct Command Accessibility**: Custom commands are now readily available within Vim.

### Purpose

The main goal is to make adding and managing custom Vim commands simpler and more intuitive, laying a foundation for future enhancements and customizations.

### Next Steps

This update paves the way for more advanced command configurations and potential integrations, offering a scalable solution to command management in Vim.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,6 +126,49 @@ require("arrow.persist").load_cache_file()
 
 Obs: persistence.nvim works fine with arrow.
 
+## Commands Module
```
Comment (otavioschwanck) [inline README.md:129]: i think this commands explanation don't belongs to the README (can confuse the new comers)

Could be on the native help system on neovim

File: lua/arrow/commands.lua [hunk 2]
```diff
@@ -0,0 +1,60 @@
+local ui = require("arrow.ui")
+
+local M = {}
+
+function M.cmd(cmd, opts)
+  local command = M.commands[cmd]
+  if command then
+    command()
```
Comment (otavioschwanck) [inline lua/arrow/commands.lua:8]: we should use opts here, right?

**Review by otavioschwanck — CHANGES_REQUESTED** [review]
Just some comments, but i liked the commands idea

File: lua/arrow/commands.lua (hunk 2 above)
Comment (tingey21) [reply lua/arrow/commands.lua:8]: I think it was yelling at me since we don't have a command that is using opts yet let me take a look

Comment (tingey21) [reply lua/arrow/commands.lua:8]: we will just have this warning lua/arrow/commands.lua|8 col 17-21 warning| This function expects a maximum of 0 argument(s) but instead it is receiving 1.

so I'll just an inspectOpts function so no warnings and then you can just get a quick look at the options

File: README.md (hunk 1 above)
Comment (tingey21) [reply README.md:129]: I'll create a new pr with contributing guide with it 👀

**Pull request merged**